### PR TITLE
Rename Inline to HorizontalStack

### LINKED ISSUE #8939: [Layout foundations] Rename `Inline` to `HorizontalStack`

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

With the UI consolidation effort happening we are aligning component names and properties across repositories. This is a first step to align component names.

### WHAT is this pull request doing?

**⚠️ Breaking change** to rename `Inline` to `HorizontalStack`

### CODE REVIEW

File: polaris.shopify.com/content/components/layout-and-structure/horizontal-stack.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Inline
+title: Horizontal Stack
```
Comment (laurkim) [inline polaris.shopify.com/content/components/layout-and-structure/horizontal-stack.md]:
```suggestion
title: Horizontal stack
```

**Review by laurkim — APPROVED** [review]

File: polaris.shopify.com/pages/examples/alpha-stack-with-inline-align.tsx [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  Page,
+  HorizontalStack,
+  Text,
+  Divider,
+} from '@shopify/polaris';
 
 import {withPolarisExample} from '../../src/components/PolarisExampleWrapper';
 
-function AlphaStackWithInlineAlignExample() {
+function AlphaStackWithHorizontalStackAlignExample() {
```
Comment (chloerice) [inline polaris.shopify.com/pages/examples/alpha-stack-with-inline-align.tsx]:
```suggestion
function AlphaStackWithInlineAlignExample() {
```

File: polaris.shopify.com/pages/examples/alpha-stack-with-inline-align.tsx [hunk 3]
```diff
@@ -57,9 +63,9 @@ const Placeholder = ({
...
           </Text>
         </div>
-      </Inline>
+      </HorizontalStack>
     </div>
   );
 };
 
-export default withPolarisExample(AlphaStackWithInlineAlignExample);
+export default withPolarisExample(AlphaStackWithHorizontalStackAlignExample);
```
Comment (chloerice) [inline polaris.shopify.com/pages/examples/alpha-stack-with-inline-align.tsx]:
```suggestion
export default withPolarisExample(AlphaStackWithInlineAlignExample);
```

**Review by chloerice — APPROVED** [review]
(Just need to revert the example name change as the "Inline" in the name refers to the `inlineAlign` prop)

**Pull request merged**